### PR TITLE
MPB Server Handler 

### PR DESCRIPTION
**Pull request by luisferreira32**
Proposal for a common handler to save us from:
* defining common return patters
* defaulting to 405 for every unimplemented method (we should actually override them to indicate which methods are available, BUT in case we forget, it's done already)

### CODE REVIEW

**Comment by luisferreira32** [comment]
FYI untested code (not sure if we need override notation or something)

**Comment by MrToino** [comment]
I will take this from here and test it on our services.

File: intmodules/log_module.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 import logging.config
 
 
-log_path = "../services/logs\\"
+log_path = "..\\logs\\"
```
Comment (luisferreira32) [inline intmodules/log_module.py:6]: change to platform agnostic paths

File: intmodules/json_validation.py [hunk 2]
```diff
@@ -17,20 +18,20 @@ def validate_schema(request):
...
 def parse_json(request):
     """Validates and parse JSON request"""
     try:
         return json.loads(request)
-    except ValueError:
-        print("Invalid request: cannot parse JSON.")
-        exit()
+    except ValueError as err:
+        logger.error(f"{err}: Invalid request: cannot parse JSON.")
+        raise InvalidJsonFormatError
```
Comment (luisferreira32) [inline intmodules/json_validation.py:34]: don't do `try except` here. if we're catching it at the server, we just need to call the functions

File: intmodules/json_validation.py [hunk 3]
```diff
@@ -17,20 +18,20 @@ def validate_schema(request):
...
         jsonschema.validate(instance=request, schema=requestSchema)
-        return True
     except ValidationError as err:
-        print("Invalid request JSON schema.")
-        exit()
-        return False
+        logger.error(f"{err}: Invalid request JSON schema.")
+        raise InvalidJsonSchemaError
+    else:
+        return True
```
Comment (luisferreira32) [inline intmodules/json_validation.py:25]: same over here, no need for try except

File: services/plotter/server.py [hunk 4]
```diff
@@ -1,37 +1,31 @@
...
+        except InvalidJsonFormatError:
+            self._return_400()
+        except InvalidJsonSchemaError:
+            self._return_400()
+        else:
+            self.send_response(200)
+            self.send_header('Content-type', 'application/json')
+            self.send_header('Content-Length', str(len(response)))
+            self.end_headers()
+            self.wfile.write(response)
```
Comment (luisferreira32) [inline services/plotter/server.py:28]: does this need to be in the `else` ? could it not be in the `try` (to be with the actual full request)

**Review by luisferreira32 — COMMENTED** [review]
Feel like it's going the right way just that we shouldn't do try excepts at the service level to only raise custom exceptions - we're only shadowing the real exceptions. However, no worries on raising custom exceptions on our own, if the logic says so (not the external libraries logic, our logic).

File: services/plotter/server.py (hunk 4 above)
Comment (MrToino) [reply services/plotter/server.py:28]: In terms of good practises, all the errors to catch will come from the `response = service(request)`, hence, the `try` statement must only contain the method that we are going in fact to try. The purpose of the `else` is exactly to run if the `try` statement was successful, so it makes sense to use it as such. Ofc we could do it all on the `try` clause but that would be considered clumsy (see [reference](https://stackoverflow.com/questions/16138232/is-it-a-good-practice-to-use-try-except-else-in-python)).
Moreover, if needed, we can also use the `finally` statement to specify code to run regardless of what happened on the `try` or `except` clauses.

**Review by MrToino — APPROVED** [review]
In this PR, we focus on the service handler itself. This review takes into consideration only changes regarding this topic. Errors/Exceptions-related changes will be re-worked on the following PRs.
This custom Service Handler is simple and it allows small modifications, easily.

**Pull request merged**